sqlmesh/core/node.py

```diff
@@ -232,23 +265,27 @@ def _inferred_interval_unit(self, sample_size: int = 10) -> IntervalUnit:
             The IntervalUnit enum.
         """
         if not self.__inferred_interval_unit:
-            croniter = CroniterCache(self.cron)
-            samples = [croniter.get_next() for _ in range(sample_size)]
-            min_interval = min(b - a for a, b in zip(samples, samples[1:]))
-            if min_interval >= 31536000:
-                self.__inferred_interval_unit = IntervalUnit.YEAR
-            elif min_interval >= 2419200:
-                self.__inferred_interval_unit = IntervalUnit.MONTH
-            elif min_interval >= 86400:
-                self.__inferred_interval_unit = IntervalUnit.DAY
-            elif min_interval >= 3600:
-                self.__inferred_interval_unit = IntervalUnit.HOUR
-            else:
-                self.__inferred_interval_unit = IntervalUnit.MINUTE
+            self.__inferred_interval_unit = interval_unit_for_cron(self.cron, sample_size)
         return self.__inferred_interval_unit
 
 
 def str_or_exp_to_str(v: t.Any) -> t.Optional[str]:
     if isinstance(v, exp.Expression):
         return v.name
     return str(v) if v is not None else None
+
+
+def interval_unit_for_cron(cron: str, sample_size: int = 10) -> IntervalUnit:
+    croniter = CroniterCache(cron)
+    samples = [croniter.get_next() for _ in range(sample_size)]
+    min_interval = min(b - a for a, b in zip(samples, samples[1:]))
+    if min_interval >= 31536000:
+        return IntervalUnit.YEAR
+    elif min_interval >= 2419200:
+        return IntervalUnit.MONTH
+    elif min_interval >= 86400:
+        return IntervalUnit.DAY
+    elif min_interval >= 3600:
```

Comment:
Can we reuse seconds that we compute above? Or better yet - can we create a factory class method in the `IntervalUnit` itself?